### PR TITLE
Abstract the keystore

### PR DESCRIPTION
**Pull request by pedropombeiro**
This PR abstracts the Keystore so that it can rely on the geth implementation or the Nimbus implementation (upcoming). It introduces two new types: `types.Key` and `types.Account` that we should use instead of the Geth versions. It also pulls in the `DecryptKey` function from geth.

### CODE REVIEW

File: account/accounts_geth.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/status-im/status-go/account/generator"
+)
+
+// GethManager represents account manager interface.
+type GethManager struct {
+	Manager
+
+	manager *accounts.Manager
```
Comment (adambabik) [inline account/accounts_geth.go]:
```suggestion
	gethAccManager *accounts.Manager
```

File: account/accounts_geth.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package account
+
+import (
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/status-im/status-go/account/generator"
+)
+
+// GethManager represents account manager interface.
+type GethManager struct {
+	Manager
```
Comment (adambabik) [inline account/accounts_geth.go]:
```suggestion
	*Manager
```
would make more sense I think. If `GethManager` was used as a value, it could produce some strange results (`Manager` will be copied).

File: account/accounts.go [hunk 3]
```diff
@@ -37,8 +35,7 @@ var zeroAddress = types.Address{}
 // Manager represents account manager interface.
 type Manager struct {
```
Comment (adambabik) [inline account/accounts.go:36]: Do we need to keep it exported?

File: account/accounts_geth.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	manager *accounts.Manager
+}
+
+// NewManager returns new node account manager.
+func NewManager() *GethManager {
+	m := &GethManager{}
+	m.accountsGenerator = generator.New(m)
```
Comment (adambabik) [inline account/accounts_geth.go]: Then here you would need to do:
```suggestion
	m.Manager = &Manager{accountsGenerator: generator.New(m)}
```

File: account/accounts_geth.go (hunk 2 above)
Comment (pedropombeiro) [reply account/accounts_geth.go]: Good point!

File: account/accounts.go (hunk 3 above)
Comment (pedropombeiro) [reply account/accounts.go:36]: For the time being yes, since it's being used in other packages like `multiaccounts` or `accounts`. If the opportunity arises we should have an interface and make the implementation private.

**Comment by pedropombeiro** [comment]
Thanks for the review Adam! I've addressed the feedback.

File: Makefile [hunk 5]
```diff
@@ -176,6 +176,10 @@ prepare-release: clean-release
 clean-release:
 	rm -rf $(RELEASE_DIR)
 
+gofmt:
```
Comment (adambabik) [inline Makefile:179]: Up to you, but there are many unrelated files changed because of this.

**Review by adambabik — APPROVED** [review]
